web-common/src/lib/time/ranges/iso-ranges.ts

```diff
@@ -0,0 +1,170 @@
+  inf: TimeRangePreset.ALL_TIME,
+};
+export function isoDurationToFullTimeRange(
+  isoDuration: string,
+  start: Date,
+  end: Date,
+  zone = "Etc/UTC"
+): TimeRange {
+  if (!isoDuration) {
+    return convertTimeRangePreset(TimeRangePreset.ALL_TIME, start, end, zone);
+  }
+  if (isoDuration in ISODurationToTimeRangePreset) {
+    return convertTimeRangePreset(
+      ISODurationToTimeRangePreset[isoDuration],
+      start,
+      end,
+      zone
+    );
+  }
+
+  const { startTime, endTime } = isoDurationToTimeRange(isoDuration, end, zone);
+  return {
+    name: TimeRangePreset.DEFAULT,
+    start: startTime,
+    end: endTime,
+  };
+}
+
+export function humaniseISODuration(isoDuration: string): string {
+  if (!isoDuration) return "";
+  const duration = Duration.fromISO(isoDuration);
+  let humanISO = duration.toHuman({
+    listStyle: "long",
+  });
+  humanISO = humanISO.replace(/(\d) (\w)/g, (substring, n, c) => {
+    return `${n} ${c.toUpperCase()}`;
+  });
+  humanISO = humanISO.replace(", and", " and");
+  return humanISO;
+}
+
+function getStartTimeTransformations(
+  isoDuration: string
+): Array<RelativeTimeTransformation> {
+  const duration = Duration.fromISO(isoDuration);
+  const period = getSmallestUnit(duration);
+  return [
+    {
+      period, // this is the offset alias for the given time range alias
+      truncationType: TimeTruncationType.START_OF_PERIOD,
+    }, // truncation
+    // then offset that by -1 of smallest period
+    {
+      duration: subtractFromPeriod(duration, period).toISO(),
+      operationType: TimeOffsetType.SUBTRACT,
+    }, // operation
+  ];
+}
+
+function getEndTimeTransformations(
+  isoDuration: string
+): Array<RelativeTimeTransformation> {
+  const duration = Duration.fromISO(isoDuration);
+  const period = getSmallestUnit(duration);
+  return [
+    {
+      duration: period,
+      operationType: TimeOffsetType.ADD,
+    },
+    {
+      period,
+      truncationType: TimeTruncationType.START_OF_PERIOD,
+    },
+  ];
+}
+
+const PeriodAndUnits: Array<{
+  period: Period;
+  unit: keyof Duration;
+}> = [
+  {
+    period: Period.MINUTE,
+    unit: "minutes",
+  },
+  {
+    period: Period.HOUR,
+    unit: "hours",
+  },
+  {
+    period: Period.DAY,
+    unit: "days",
+  },
+  {
+    period: Period.WEEK,
+    unit: "weeks",
+  },
+  {
+    period: Period.MONTH,
+    unit: "months",
+  },
+  {
+    period: Period.YEAR,
+    unit: "years",
+  },
+];
+const PeriodToUnitsMap: Partial<Record<Period, keyof Duration>> = {};
+PeriodAndUnits.forEach(({ period, unit }) => (PeriodToUnitsMap[period] = unit));
+
+function getSmallestUnit(duration: Duration) {
+  for (const { period, unit } of PeriodAndUnits) {
+    if (duration[unit]) {
+      return period;
+    }
+  }
+
+  return undefined;
+}
+
+function subtractFromPeriod(duration: Duration, period: Period) {
+  if (!PeriodToUnitsMap[period]) return duration;
+  return duration.minus({ [PeriodToUnitsMap[period]]: 1 });
+}
```

Comment:
Since this is a transformation function independent of ISOs, I would move this to `transforms/index`